lib/permutations.rb

```diff
@@ -1,4 +1,26 @@
 
 def permutations?(string1, string2)
-  raise NotImplementedError, "permutations? not implemented"
+  intersections = {}
+  string1_array = string1.split("")
+  string2_array = string2.split("")
+
+  string1_array.each do |value|
+    if !intersections[value]
+      intersections[value] = 1
+    elsif intersections[value]
+      intersections[value] += 1
+    end
+  end
+
+  string2_array.each do |value|
+    if intersections[value]
+      intersections[value] += 1
+    end
```

Comment:
What if the value isn't in `intersections`?  Why not return false, if it's not there.

I would also suggest subtracting instead of adding, and then you can check to see if all the values are 0.